### PR TITLE
fix formatting with value fallback, add basic test

### PR DESCRIPTION
**Pull request by pickx**
`#[derive(BinaryBits)]` did not account for the case of a value fallback.

### CODE REVIEW

**Review by hecatia-elegua — APPROVED** [review]
Thank you!

File: tests/fmt.rs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    reserved: u13,
+    reg1: bool,
+    reg2: u16,
+    reserved: u4,
+    reg3: u18,
+}
+
+#[bitsize(64)]
+#[derive(FromBits, BinaryBits)]
+struct Lunch(Bangers, Mash, Register);
```
[the remark this answers is not in the document]
Comment (pickx) [reply tests/fmt.rs:34]: registers, a staple of british cuisine

File: tests/fmt.rs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        "0b00"
+    );
+    
+    // this one has underscores
+    assert_eq!(
+        format!("0b{:b}", lunch.val_2()),
+        "0b110010110010101001_1011_1011011001100011_1_1011001100000"
+    );
+
+    // ...but the underscores are not "inherited" for the nested struct
```
[the remark this answers is not in the document]
Comment (pickx) [reply tests/fmt.rs:62]: this seems possible to implement, but is a hassle.

File: tests/fmt.rs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    let register = Register::from(u52::new(0b110010110010101001_1011_1011011001100011_1_1011001100000));
+    let b = Bangers::from(u10::new(0b1100110011));
+    let m = Mash::from(<u2 as Number>::MIN);
+    
+    let lunch = Lunch::new(b, m, register);
+
+    // fallback value is used
+    assert_eq!(
+        format!("0b{:b}", lunch.val_0()),
+        "0b1100110011"
```
[the remark this answers is not in the document]
Comment (pickx) [reply tests/fmt.rs:47]: thanks, added

**Review by hecatia-elegua — APPROVED** [review]

File: tests/fmt.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (pickx) [reply tests/fmt.rs]: ...though it does show the inconsistency with `std` (we pad with zeroes, `u16` doesn't)

**Review by hecatia-elegua — APPROVED** [review]

**Pull request merged**